lib/ssl/src/ssl_certificate.erl

```diff
@@ -326,6 +328,56 @@ handle_cert_auths([_ | Certs] = EChain, CertAuths, _, _) ->
             {error, EChain, not_in_auth_domain}
     end.
 
+available_cert_key_pairs(CertKeyGroups) ->
+    RevAlgos = [dsa, rsa, rsa_pss_pss, ecdsa, eddsa],
+    cert_key_group_to_list(RevAlgos, CertKeyGroups, []).
+
+available_cert_key_pairs(CertKeyGroups, Version) ->
+    RevAlgos = [dsa, rsa, rsa_pss_pss, ecdsa, eddsa],
+    cert_key_group_to_list(RevAlgos, CertKeyGroups, Version, []).
+
+cert_key_group_to_list([], _, Acc) ->
+    final_group_list(Acc);
+cert_key_group_to_list([Algo| Rest], CertKeyGroups, Acc) ->
```

Comment:
Need an explanation if this whole function and what it is supposed to do.

Reply:
The inputted list of cert and key pairs is in arbitrary order and may represent all or some of the possible signing algorithms. Internally this list will be grouped in different lists (one per signing algorithm where each list can be sorted on the relative key strength unless it is an engine key). Then later we will use these groups the create one prioritized list that we can use the first one supported by negotiate algorithms and other relevant criteria.